packages/interchain-token-service-api/src/primitives.rs

```diff
@@ -0,0 +1,90 @@
+use crate::error::Error;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{HexBinary, Uint256};
+use error_stack::Report;
+
+#[cw_serde]
+#[derive(Eq)]
+pub struct TokenId {
+    #[serde(with = "axelar_wasm_std::hex")]
+    #[schemars(with = "String")]
+    pub id: [u8; 32],
+}
+
+#[cw_serde]
+#[derive(Eq, Copy)]
+#[repr(u8)]
+pub enum TokenManagerType {
+    NativeInterchainToken = 0,
+    MintBurnFrom = 1,
+    LockUnlock = 2,
+    LockUnlockFee = 3,
+    MintBurn = 4,
+    Gateway = 5,
+}
+
+#[cw_serde]
+#[derive(Eq)]
+pub enum ITSMessage {
+    InterchainTransfer {
+        token_id: TokenId,
+        source_address: HexBinary,
+        destination_address: HexBinary,
+        amount: Uint256,
+        data: HexBinary,
```

Comment:
Is this the full payload or the hash?

Reply:
Full payload. The whole struct gets encoded into the GMP payload that we approve over. But because ITS hub needs to receive the GMP payload to process, the ITS contract data also gets included. While we don't store this payload, it is going to affect p2p bandwidth and get included into the block data unfortunately. Not including this would require breaking changes on the relayers to support ITS GMP execute calls in a special way.